internal/pkg/deploy/cloudformation/env.go

```diff
@@ -52,12 +52,12 @@ func (cf CloudFormation) StreamEnvironmentCreation(env *deploy.CreateEnvironment
 }
 
 // DeleteEnvironment deletes the CloudFormation stack of an environment.
-func (cf CloudFormation) DeleteEnvironment(appName, envName string) error {
+func (cf CloudFormation) DeleteEnvironment(appName, envName, cfnExecRoleARN string) error {
 	conf := stack.NewEnvStackConfig(&deploy.CreateEnvironmentInput{
 		AppName: appName,
 		Name:    envName,
 	})
-	return cf.cfnClient.DeleteAndWait(conf.StackName())
+	return cf.cfnClient.DeleteAndWaitWithRoleARN(conf.StackName(), cfnExecRoleARN)
```

Comment:
Are we allowed to use CFN exec role to delete the env stack if it is created by profile role and never gets updated by CFN exec role?

Reply:
yup, you don't need to update w the role first, you can just delete with the role immediately. It's just that once you use specify a role for CFN then you can't remove it:
> It is not possible to remove a service role attached to a stack after the stack is created.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-iam-servicerole.html